src/AudioInputsService.ts

```diff
@@ -18,38 +14,96 @@ type AudioInputsServiceEvents = {
 };
 
 class AudioInputsService extends EventEmitter<AudioInputsServiceEvents> {
-  private timeoutId: number | undefined;
+  private timeoutId?: number;
+
+  private microphone?: MediaRecorder;
+
+  private socket?: Socket;
 
-  private microphone: MediaRecorder | undefined;
+  private streamTimeslice: number;
 
-  private socket: Socket | undefined;
+  private ready = false;
 
-  public startListening = async ({
-    timeout = 10000,
-  }: SpeechRecognitionOptions = {}): Promise<void> => {
-    if (!this.microphone) {
-      const userMedia = await navigator.mediaDevices.getUserMedia({
-        audio: true,
+  constructor(streamTimeslice: number | undefined) {
+    super();
+
+    this.streamTimeslice = streamTimeslice ?? 100;
+  }
+
+  public connect = (token: string): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      if (this.socket) {
+        console.log("Socket already connected");
+        resolve();
+      }
+
+      this.socket = io(STT_URL, {
+        transports: ["websocket"],
+        query: { token },
       });
 
-      this.microphone = new MediaRecorder(userMedia);
-    }
+      this.socket.on("error", (error: string) => {
+        console.error(error);
+        this.emit("error", error);
+        reject(error);
+      });
 
-    this.microphone.start(100);
+      this.socket.on("transcript", (transcript: string) => {
+        console.log("Received transcript:", transcript);
+        if (transcript) {
+          this.emit("transcript", transcript);
+        }
+      });
 
-    if (this.timeoutId !== undefined) {
-      clearTimeout(this.timeoutId);
+      this.socket.on("connect", () => {
+        // Deepgram requires a short interval before data is sent.
+        setTimeout(() => {
+          this.ready = true;
+          resolve();
+        }, 2000);
+      });
+    });
+  };
+
+  public startListening = async (timeout = 10000): Promise<void> => {
+    try {
+      if (!this.ready) {
+        return;
+      }
+
+      if (!this.microphone) {
+        const userMedia = await navigator.mediaDevices.getUserMedia({
```

Comment:
can we put this in a separate private setupMicrophone method to cut some length from startListening which has a lot going on?